sae_lens/training/sae_trainer.py

```diff
@@ -84,7 +90,9 @@ def __init__(
                 range(
                     0,
                     cfg.total_training_samples,
-                    cfg.total_training_samples // self.cfg.n_checkpoints,
+                    math.ceil(
+                        cfg.total_training_samples / (self.cfg.n_checkpoints + 1)
```

Comment:
There was some wierdness happening when `total_training_samples` was not exactly divisible by `n_checkpoints`, where you could end up with an extra checkpoint. This should fix that issue.